### PR TITLE
fix: fixed hanging pod termination

### PR DESCRIPTION
**Pull request by maximilianwerk**
This PR fixed the hanging pods bug. It also reintroduces the integration test for dump reload and demonstrate its usage.

### CODE REVIEW

File: jina/peapods/zmq/__init__.py [hunk 1]
```diff
@@ -495,7 +495,11 @@ def send_message(
     num_bytes = 0
     try:
         _prep_send_socket(sock, timeout)
-        sock.send_multipart(msg.dump())
+        if 'flags' in kwargs:
```
Comment (JoanFM) [inline jina/peapods/zmq/__init__.py:498]: we still have no reason for this right?

Comment (maximilianwerk) [reply jina/peapods/zmq/__init__.py:498]: @florian-hoenicke said there were some issues, when not setting it. I can remove it and we see.

**Comment by nan-wang** [comment]
reference: https://pyzmq.readthedocs.io/en/latest/api/zmq.html#zmq.Socket.send

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Comment by hanxiao** [comment]
Im reverting this PR as im not convinced by adding NOBLOCK flag, seems without solid reason. Playing network parameter in this way is dangerous and should be avoided.

**Pull request merged**